apps/meteor/app/livechat/server/lib/Contacts.ts

```diff
@@ -1,16 +1,28 @@
+import { ILivechatCustomField, ILivechatVisitor, ILivechatVisitorDTO, IOmnichannelRoom } from '@rocket.chat/core-typings';
 import { check } from 'meteor/check';
 import s from 'underscore.string';
 
-import { LivechatVisitors, LivechatCustomField, LivechatRooms, Rooms, LivechatInquiry, Subscriptions } from '../../../models';
+import { LivechatVisitors, LivechatCustomField, LivechatRooms, Rooms, LivechatInquiry, Subscriptions } from '../../../models/server';
 
 export const Contacts = {
-	registerContact({ token, name, email, phone, username, customFields = {}, contactManager = {} } = {}) {
+	registerContact({
+		token,
+		name,
+		email,
+		phone,
+		username,
+		customFields = {},
+		contactManager,
+	}: Pick<ILivechatVisitorDTO, 'token' | 'name' | 'email' | 'username' | 'phone'> & {
+		customFields?: Record<string, unknown>;
+		contactManager?: ILivechatVisitor['contactManager'];
+	}): string {
 		check(token, String);
 
-		const visitorEmail = s.trim(email).toLowerCase();
+		const visitorEmail = email && s.trim(email).toLowerCase();
 
 		let contactId;
-		const updateUser = {
+		const updateUser: any = {
```

Comment:
😬

Reply:
I believe you can use one of the mongo types (like `UpdateQuery` for representing this (and avoid `any`)